interactive/dashboard/api.py

```diff
@@ -0,0 +1,1036 @@
+                             'configs/'+request.session.session_key)        
+        if not os.path.isdir(direc):
+            log("* detected no data from this user")
+            return Response(status=status.HTTP_200_OK)
+        else:
+            # path to save data for user
+            if not 'index' in request.GET:
+                return JsonResponse({"error": "missing reaction index"})
+            conf = ('configs/'+request.session.session_key
+                    +"/camp_data/reactions.json")
+            config_path = os.path.join(
+                settings.BASE_DIR, conf)
+            reaction_detail = reactions_info(
+                config_path)[int(request.GET['index'])]
+            reaction_detail['index'] = int(request.GET['index'])
+            return JsonResponse(reaction_detail)
+
+
+class RemoveReactionView(views.APIView):
+    def get(self, request):
+        log("****** GET request received REMOVE_REACTION ******")
+        # print("current cookie:" + request.COOKIES['sessionid'])
+        if not request.session.session_key:
+            request.session.create()
+        if not 'index' in request.GET:
+            return HttpResponseBadRequest("missing reaction index")
+        direc = os.path.join(settings.BASE_DIR,
+                             'configs/'+request.session.session_key)
+        # check for config dir
+        if not os.path.isdir(direc):
+            log("* detected no data from this user")
+            return Response(status=status.HTTP_200_OK)
+        else:
+            conf = ('configs/'+request.session.session_key
+                    +"/camp_data/reactions.json")
+            config_path = os.path.join(
+                settings.BASE_DIR, conf)
+            reaction_remove(int(request.GET['index']), config_path)
+            return HttpResponse('')
+
+
+class SaveReactionView(views.APIView):
+    def get(self, request):
+        print("****** GET request received ADD_REACTION ******")
+        # print("current cookie:" + request.COOKIES['sessionid'])
+        return JsonResponse({"error": 
+                             "saving a reaction should be a POST request"})
+
+    def post(self, request):
+        if not request.session.session_key:
+            request.session.create()
+
+        reaction_data = json.loads(request.body)
+        conf = ('configs/'+request.session.session_key
+                    +"/camp_data/reactions.json")
+        config_path = os.path.join(
+            settings.BASE_DIR, conf)
+        reaction_save(reaction_data, config_path)
+        return JsonResponse({})
+
+
+class ReactionTypeSchemaView(views.APIView):
+    def get(self, request):
+        log("****** GET request received SCHEMATYPEVIEW ******")
+        # print("current cookie:" + request.COOKIES['sessionid'])
+        if not request.session.session_key:
+            request.session.create()
+        if not 'type' in request.GET:
+            return JsonResponse({"error": "missing reaction type"})
```

Comment:
BadRequest, but see conversation above

Reply:
used to actually give error message back to user

Reply:
I suppose, but it's much simpler for someone working on this code later on that it's sending back an error for missing reaction type, if I added it to the request body it would be like 2-3 more lines of code and a bit harder to understand. at least that was my thinking, i may be wrong

Reply:
I am saying that you should return a BadRequest *with* this json object so that the error can be seen.

Reply:
Future work for #291